### PR TITLE
Update libnetwork version for IPv6 support

### LINKED ISSUE #28055: The --ipv6 flag to docker network create is broken for remote plugins in Docker 1.12.2
**Description**
If I run `docker network create` with a remote network driver and the --ipv6 flag, I expect my remote plugin to be passed the "V6" flag.

**Steps to reproduce the issue:**
1. Run dockerd with a cluster store configured - `sudo dockerd --cluster-store=etcd://127.0.0.1:2379`
2. Run etcd locally `docker run --detach --net=host quay.io/coreos/etcd etcd --advertise-client-urls "http://localhost:2379" --listen-client-urls "http://0.0.0.0:2379"`
3. Run the Calico Libnetwork plugin - `docker run --rm --net=host --privileged -e CALICO_ETCD_AUTHORITY=localhost:2379 -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock calico/libnetwork-plugin:v1.0.0-beta-rc1 /libnetwork-plugin
4. Create a network -  `docker network create --ipv6=true  --driver=calico --ipam-driver=calico-ipam ipv6`

**Describe the results you received:**
Calico logs:
```
2016/11/04 01:30:34 GetCapabilities response JSON={"Scope":"global"}
2016/11/04 01:30:34 GetCapabilities response JSON={"RequiresMACAddress":false}
2016/11/04 01:30:34 GetDefaultAddressSpace response JSON={"LocalDefaultAddressSpace":"CalicoLocalAddressSpace","GlobalDefaultAddressSpace":"CalicoGlobalAddressSpace"}
2016/11/04 01:30:34 RequestPool JSON={"AddressSpace":"CalicoGlobalAddressSpace","Pool":"","SubPool":"","Options":{},"V6":false}
2016/11/04 01:30:34 RequestPool response JSON={"PoolID":"CalicoPoolIPv4","Pool":"0.0.0.0/0","Data":{"com.docker.network.gateway":"0.0.0.0/0"}}
2016-11-04 01:30:34.556470 I | Entering go-plugins-helpers createnetwork
2016/11/04 01:30:34 CreateNetwork JSON={"NetworkID":"20a4483125b4e7af19f370d12799ddfbb7ceb05dec4c9e89739db2b42135641b","Options":{"com.docker.network.enable_ipv6":true,"com.docker.network.generic":{}},"IPv4Data":[{"AddressSpace":"CalicoGlobalAddressSpace","Pool":"0.0.0.0/0","Gateway":"0.0.0.0/0","AuxAddresses":null}],"IPv6Data":[]}
2016/11/04 01:30:34 CreateNetwork response JSON={}
```

There are no logs from the Docker daemon

**Describe the results you expected:**
`"V6":false` should be `"V6":true` on the RequestPool

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
docker version
Client:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 18:29:41 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 18:29:41 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
docker info
Containers: 6
 Running: 2
 Paused: 0
 Stopped: 4
Images: 45
Server Version: 1.12.2
Storage Driver: overlay2
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: calico bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.8.2-040802-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.25 GiB
Name: xeon
ID: CM6N:TGLQ:WIRF:5JGO:BX7D:ZPLC:GBBO:WEHD:U6C4:3PCD:WODP:T6FS
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: tomdee
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Cluster Store: etcd://localhost:2379
Insecure Registries:
 127.0.0.0/8

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by tomdee**
Fixes #28055 
Also fixes two bugs which would cause a panic in `sandbox.go` and `osl/neigh_linux.go`

### CODE REVIEW

**Comment by aboch** [comment]
@docker/tools-team 
Regarding the test failure

>00:47:37 FAIL: docker_cli_network_unix_test.go:714: DockerNetworkSuite.TestDockerNetworkInspectCustomSpecified
00:47:37 
00:47:37 docker_cli_network_unix_test.go:715:
00:47:37     dockerCmd(c, "network", "create", "--driver=bridge", "--ipv6", "--subnet=172.28.0.0/16", "--ip-range=172.28.5.0/24", "--gateway=172.28.5.254", "br0")


simply add and IPv6 `--subnet` to the command.

**Comment by aboch** [comment]
@tomdee
Since you need to update your PR, please vendor libnetwork up to https://github.com/docker/libnetwork/commit/f36e733a08cd8239a2db296994cb6613fef1cece 

Thanks!

**Comment by tomdee** [comment]
@aboch Can this be added to the v1.13 milestone?

**Comment by mavenugo** [comment]
@tomdee pls update the vendoring to https://github.com/docker/libnetwork/commit/f36e733a08cd8239a2db296994cb6613fef1cece so that we can easily cherry-pick this PR into the 1.3 branch.

**Comment by tomdee** [comment]
@mavenugo @aboch I tried to update to the docker/libnetwork@f36e733 but got the following

```
# github.com/docker/docker/vendor/github.com/docker/libnetwork/ns
vendor/github.com/docker/libnetwork/ns/init_linux.go:36: initNl.SetSocketTimeout undefined (type *netlink.Handle has no field or method SetSocketTimeout)
```

I suggest you update the vendoring for that in a follow on PR

**Comment by aboch** [comment]
@tomdee Ah yes, we also need to vendor latest netlink

**Comment by tomdee** [comment]
I've updated the test and it's passing locally.

**Comment by tomdee** [comment]
@aboch  @mavenugo  Pushed with netlink and libnetwork vendors. And the tests are updated.

I'd love to see this merged and cherry picked in 1.13 ASAP, but I'm on vacation until after thanksgiving so can one you handle that?

File: integration-cli/docker_cli_network_unix_test.go [hunk 1]
```diff
@@ -712,7 +712,7 @@ func (s *DockerNetworkSuite) TestDockerNetworkInspectCustomUnspecified(c *check.
 }
 
 func (s *DockerNetworkSuite) TestDockerNetworkInspectCustomSpecified(c *check.C) {
-	dockerCmd(c, "network", "create", "--driver=bridge", "--ipv6", "--subnet=172.28.0.0/16", "--ip-range=172.28.5.0/24", "--gateway=172.28.5.254", "br0")
+  dockerCmd(c, "network", "create", "--driver=bridge", "--ipv6", "--subnet=fd80:24e2:f998:72d6::/64", "--subnet=172.28.0.0/16", "--ip-range=172.28.5.0/24", "--gateway=172.28.5.254", "br0")
```
Comment (runcom) [inline integration-cli/docker_cli_network_unix_test.go]: Indent seems wrong here

**Comment by aboch** [comment]
Thanks @tomdee 
changes look good to me

**Review by vdemeester — APPROVED** [review]
LGTM :frog:

**Comment by mavenugo** [comment]
LGTM

**Comment by mavenugo** [comment]
@sanimej @aboch we have to vendor in libnetwork to the latest and then cherry-pick into 1.13.0 bump branch. can u please take care of that ?

**Comment by aboch** [comment]
Also fixes #25184

**Pull request merged**